apps/dashboard/app/helpers/recently_used_apps_helper.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+# Helper methods for RecentlyUsedApps widget
+module RecentlyUsedAppsHelper
+  # Returns the 4 most recently used interactive applications based on their cache file
+  # @return [Array<BatchConnect::App>] The 4 most recently used apps
+  def recently_used_apps
+    cache_files = BatchConnect::Session.cache_root.children.sort_by do |pathname|
```

Comment:
This looks expensive to compute. Maybe we should put this within a cache block.

```ruby
Rails.cache.fetch('recently_used_apps', expires_in: 1.hour) do

end
```

There could be more effective caching schemes, i.e., invalidate it when one of the cache files is updated, but I feel like we can defer that.

Reply:
Added caching.